contracts/cw-distributor/src/contract.rs

```diff
@@ -0,0 +1,130 @@
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{to_binary, CosmosMsg, Uint128, WasmMsg};
+
+use cosmwasm_std::{Binary, Deps, DepsMut, Env, MessageInfo, Response, StdResult};
+use cw2::set_contract_version;
+use stake_cw20::msg::ReceiveMsg;
+
+use crate::error::ContractError;
+use crate::msg::{ConfigResponse, ExecuteMsg, InstantiateMsg, LastPaymentBlockResponse, QueryMsg};
+use crate::state::{Config, CONFIG, LAST_PAYMENT_BLOCK};
+
+const CONTRACT_NAME: &str = "crates.io:cw-distributor";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    env: Env,
+    _info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+    let owner = deps.api.addr_validate(&msg.owner)?;
+    let recipient = deps.api.addr_validate(&msg.recipient)?;
+
+    let reward_token = deps.api.addr_validate(&msg.reward_token)?;
```

Comment:
Might be nice to run some queries against these addresses when we set them to verify that these are indeed token / staking contracts. Could run `TotalStakedAtHeight` and `TokenInfo`.

Reply:
@ben2x4 Lmk if you think this type of validation is necessary. I'm down to add it

Reply:
hmm, I know @ezekiiel  likes these types of checks, but i don't have a strong preference.

Reply:
Added these checks + tests